disposebag/src/test/kotlin/com/kizitonwose/android/disposebag/DisposeBagTest.kt

```diff
@@ -33,50 +36,60 @@ class DisposeBagTest {
     }
 
     @Test
-    fun disposedByDisposeBag() {
-        val disposeBag = DisposeBag(lifecycleOwner)
-
+    fun shouldDisposeOnDefaultEndEvent() {
+        //Arrange
         val emittedItems = mutableListOf<Long>()
 
+        //Act
         val disposable = Observable.interval(5, TimeUnit.SECONDS, testScheduler)
                 .subscribe {
                     emittedItems.add(it)
-                    System.out.println(it)
                 }.apply { disposedBy(disposeBag) }
 
-        assertTrue(emittedItems.isEmpty())
+        //Assert
+        assertEmptyList(emittedItems)
 
         testScheduler.advanceTimeBy(20, TimeUnit.SECONDS)
 
-        assertTrue(emittedItems.size == 4)
+        assertListHasSize(emittedItems, 4)
 
         lifecycleOwner.performDestroy()
 
         assertTrue(disposable.isDisposed)
 
         testScheduler.advanceTimeBy(10, TimeUnit.SECONDS)
 
-        assertTrue(emittedItems.size == 4)
+        assertListHasSize(emittedItems, 4)
     }
 
     @Test
-    fun disposeMultipleByDisposeBagWithCustomEvent() {
+    fun shouldDisposeAllDisposablesWhenEndEventOccurs() {
+        //Arrange
         lifecycleOwner.lifecycleRegistry.markState(Lifecycle.State.STARTED)
+        val emmittedItems = mutableListOf<Long>()
 
+        //Act
         val disposable1 = Observable.interval(5, TimeUnit.SECONDS, testScheduler)
                 .subscribe {
-                    System.out.println(it)
+                    emmittedItems.add(it)
                 }
 
-        val disposable2 = Observable.fromIterable(1..100)
-                .delay(10, TimeUnit.SECONDS)
+        val disposable2 = Observable.fromIterable(1..3)
+                .delay(10, TimeUnit.SECONDS, testScheduler)
                 .subscribe {
-                    System.out.println(it)
+                    emmittedItems.add(it.toLong())
                 }
 
-        val disposeBag = DisposeBag(listOf(disposable1, disposable2),
+        disposeBag = DisposeBag(listOf(disposable1, disposable2),
                 lifecycleOwner, Lifecycle.Event.ON_STOP)
 
+        assertEmptyList(emmittedItems)
+
+        testScheduler.advanceTimeBy(5 , TimeUnit.SECONDS)
+
+        assertListHasSize(emmittedItems, 1)
```

Comment:
This test is not meant to check that items are emitted, that is already covered. This is only meant to check that the disposable is properly disposed.

Reply:
I understand your point but you still to be exhaustive, event hough that is not what we explicitly testing for, I think just for the purpose of correctness this still needs to checked.